Sources/SwiftParser/SyntaxUtils.swift

```diff
@@ -88,6 +88,10 @@ extension SyntaxText {
   var isEditorPlaceholder: Bool {
     return self.starts(with: SyntaxText("<#")) && self.hasSuffix(SyntaxText("#>"))
   }
+
+  var isStartingWithSuppercase: Bool {
+    return description.first?.isUppercase ?? false
```

Comment:
I don’t think we need to render the entire buffer into a `Swift.String` to decide whether the first character is uppercase. You should be able to do the following based on the ASCII value of the first character.

```swift
if let firstCharacterByte = buffer.first {
  return 65 <= firstCharacterByte && firstCharacterByte <= 90
}
```

---

Also there’s a typo. But I like `Suppercase` 🍲

Reply:
It could also be moved, but I'm not sure if that's correct tough. It doesn't quite fit I think.